### PR TITLE
HMA-5087 InformationMessageCardView

### PR DESCRIPTION
**Pull request by emmaatkins**
# 📝 Description

Github Issue
HMA-5087 InformationMessageCardView
  
- [ x] Updated CHANGELOG
- [ ] Updated README

# 🛠 Testing Notes
New Compose InfomationMessageCardView component added
Note: button onClick does not produce an action or Toast message

# 📸 Screenshots
IMCV1
IMCV2

### CODE REVIEW

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/atom/button/Button.kt [hunk 1]
```diff
@@ -98,29 +101,31 @@ fun PrimaryButton(
...
-            buttonColors = buttonColors,
-            onClick = onClick,
-        ) {
-            content()
+object SecondaryButton {
+    @Composable
+    operator fun invoke(
+        text: String,
+        modifier: Modifier = Modifier,
+        enabled: Boolean = true,
```
Comment (jemmaSlater) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/atom/button/Button.kt:109]: I don't think we need to define this here, so we may as well tidy up now we're here. The `enabled` field now has a default value of true in `HmrcButton` and there isn't a use case where we would want to have a disabled secondary button, so we can tidy up and remove this, and just rely on the default that's built in.
```suggestion
```

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/atom/button/Button.kt [hunk 2]
```diff
@@ -98,29 +101,31 @@ fun PrimaryButton(
...
+        ),
+        onClick: () -> Unit,
+        content: @Composable RowScope.() -> Unit = {},
+    ) {
+        CompositionLocalProvider(LocalRippleTheme provides HmrcRippleTheme) {
+            HmrcButton(
+                text = text,
+                modifier = modifier,
+                contentPadding = PaddingValues(all = HmrcTheme.dimensions.hmrcSpacing16),
+                enabled = enabled,
```
Comment (jemmaSlater) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/atom/button/Button.kt:122]:
```suggestion
```

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+    headline: String,
+    headlineContentDescription: String = "",
+    text: String? = null,
+    messageType: MessageType,
+    buttons: List<InformationMessageButton> = emptyList()
+) {
+    val textColor: Color by remember { mutableStateOf(messageType.headlineTint) }
+    val iconColor: Color by remember { mutableStateOf(messageType.iconColor) }
+    val backgroundColor: Color by remember { mutableStateOf(messageType.backgroundColor) }
+    val icon: Int by remember { mutableIntStateOf(messageType.icon) }
```
Comment (jemmaSlater) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:79]: Is there any benefit having these defined here? Could we not just use the values directly from the parameters as they are?

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                painter = painterResource(id = icon),
+                contentDescription = "",
+                colorFilter = ColorFilter.tint(iconColor)
+            )
+            Spacer(modifier = Modifier.width(HmrcTheme.dimensions.hmrcSpacing8))
+            Text(
+                text = headline,
+                modifier = Modifier
+                    .weight(1f)
+                    .heightIn(HmrcTheme.dimensions.hmrcIconSize36)
```
Comment (jemmaSlater) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:99]: Do we need this? Would this restrict the size of the content if we had a lot of text?

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/ui/theme/Color.kt [hunk 5]
```diff
@@ -45,3 +46,4 @@ val HmrcGrey2Dark = Color(0xFF6F777B)
 val HmrcGrey3Dark = Color(0xFF0B0C0C)
 val HmrcPinkDark = Color(0xFFBB94FF)
 val HmrcYellowDark = Color(0xFFFEFF4F)
+val HmrcAlwaysBlackDark = Color(0xFF0B0C0C)
```
Comment (jemmaSlater) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/ui/theme/Color.kt:49]: Do we need this? Is it used anywhere?

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                headline = stringResource(id = R.string.info_message_placeholder_headline),
+                text = stringResource(id = R.string.info_message_placeholder_headline),
+                messageType = MessageType.URGENT,
+                buttons = listOf(InformationMessageButton("Headline button", MessageType.URGENT, onClick = {}))
+            )
+        }
+
+        ExamplesSlot {
+            // Example 1
+            HmrcCardView(modifier = Modifier.padding(bottom = HmrcTheme.dimensions.hmrcSpacing16)) {
```
Comment (jemmaSlater) [inline sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt:47]: I don't think we need this card view.. the component has one built in. Same comment applies to the examples below.
```suggestion
```

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+                        bottom = HmrcTheme.dimensions.hmrcSpacing16
+                    )
+            ) {
+                Text(
+                    text = text,
+                    modifier = Modifier,
+                    style = HmrcTheme.typography.body,
+                    color = textColor
+                )
+            }
```
Comment (aliahmedbd) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:124]: Can we remove the parent Row here?

```suggestion
              Text(
                text = text,
                modifier = modifier
                    .fillMaxWidth()
                    .padding(
                        start = HmrcTheme.dimensions.hmrcSpacing16,
                        end = HmrcTheme.dimensions.hmrcSpacing16,
                        bottom = HmrcTheme.dimensions.hmrcSpacing16
                    ),
                style = HmrcTheme.typography.body,
                color = textColor
            )
```

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+                    ),
+                    null
+                )
+            }
+            ButtonType.OUTLINE -> {
+                Pair(
+                    ButtonDefaults.buttonColors(
+                        containerColor = Color.Transparent, contentColor = messageType.headlineTint
+                    ),
+                    BorderStroke(width = 1.0.dp, color = messageType.headlineTint,)
```
Comment (aliahmedbd) [inline components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:158]: Do we need to define the button stroke on the `Dimensions` file?

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+import uk.gov.hmrc.components.compose.organism.information.InformationMessageCardView
+import uk.gov.hmrc.components.compose.organism.information.MessageType
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+import uk.gov.hmrc.sample_compose_components.R
+import uk.gov.hmrc.sample_compose_fragments.presentation.screens.sampletemplate.ExamplesSlot
+import uk.gov.hmrc.sample_compose_fragments.presentation.screens.sampletemplate.PlaceholderSlot
+import uk.gov.hmrc.sample_compose_fragments.presentation.screens.sampletemplate.ScreenScrollViewColumn
+
+@Composable
+fun InformationMessageCardViewScreen(onClickAction: () -> Unit) {
```
Comment (aliahmedbd) [inline sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt:34]: It appears that `onClickAction: () -> Unit` is not being used.

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/fragment/organisms/InformationMessageCardViewFragment.kt [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = FragmentComposeExampleBinding.bind(view)
+        binding.composeView.apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                val window = rememberWindowSizeClass()
+                HmrcTheme(window) {
+                    HmrcSurface {
+                        InformationMessageCardViewScreen { showToast(getString(R.string.cta_pressed)) }
```
Comment (aliahmedbd) [inline sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/fragment/organisms/InformationMessageCardViewFragment.kt:44]: As click action has no usage on `InformationMessageCardViewScreen` that's `showToast(getString(R.string.cta_pressed))` is not invoked anymore. Do we need this?

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt (hunk 4 above)
Comment (emmaatkins) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:99]: This is taken from the WarningView and it doesn't display the icon/text correctly without it.

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/fragment/organisms/InformationMessageCardViewFragment.kt (hunk 10 above)
Comment (emmaatkins) [reply sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/fragment/organisms/InformationMessageCardViewFragment.kt:44]: Removed

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt (hunk 9 above)
Comment (emmaatkins) [reply sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt:34]: Removed

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+                )
+            }
+        }
+        CompositionLocalProvider(LocalRippleTheme provides HmrcRippleTheme) {
+            HmrcButton(
```
[the remark this answers is not in the document]
Comment (emmaatkins) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:163]: @jemmaSlater thoughts?

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt (hunk 7 above)
Comment (emmaatkins) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:124]: This breaks it

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt (hunk 8 above)
Comment (emmaatkins) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:158]: That doesn't seem to change anything?

Comment (aliahmedbd) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:158]: Yes, that's correct. There's nothing to alter; it's likely that the rendering results between XML and Compose are slightly different.

File: components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt (hunk 11 above)
[the remark this answers is not in the document]
Comment (jemmaSlater) [reply components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt:163]: I think we can keep it contained to this use case for now, but abstract later if we find there are more usages.

File: sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+
+            Spacer(modifier = Modifier.height(HmrcTheme.dimensions.hmrcSpacing16))
+
+            // Example 4
+                InformationMessageCardView(
+                    headline = stringResource(id = R.string.info_message_example_4_headline),
+                    headlineContentDescription = stringResource(id = R.string.info_message_example_4_headline_content_description),
+                    messageType = MessageType.NOTICE
+                )
+        }
```
Comment (jemmaSlater) [inline sample-compose-fragments/src/main/java/uk/gov/hmrc/sample_compose_fragments/presentation/screens/organisms/InformationMessageCardViewScreen.kt:88]: I think you forgot to update the indentation after removing the cards here.. can we tidy it up a bit and re-indent?

**Review by jemmaSlater — DISMISSED** [review]

**Review by jemmaSlater — APPROVED** [review]

**Review by aliahmedbd — DISMISSED** [review]

**Review by aliahmedbd — APPROVED** [review]

**Pull request merged**